lib/iknow_params/serializer.rb

```diff
@@ -95,8 +96,11 @@ def initialize
       super(::Integer)
     end
 
-    def load(str)
-      Integer(str)
+    # JSON only supports floats, so we have to accept a value
+    # which may have already been parsed into a Ruby Float or Integer.
```

Comment:
Given that JSON only supports floats, and that Ruby's JSON parser picks a type for them based only on their syntax (`JSON.parse("[1, 1.0]").map(&:class) => [Fixnum, Float]`), is there any chance of the input coming in in a normalized form? (i.e. `1.0` to `1` or vice versa)